src/Algebra/Construct/Flip/Op.agda

```diff
@@ -163,6 +171,23 @@ module _ {≈ : Rel A ℓ} {∙ : Op₂ A} where
     }
     where module g = IsAbelianGroup g
 
+module _ {≈ : Rel A ℓ} {+ * : Op₂ A} { - : Op₁ A} {0# 1# : A} where
+
+  isRing : IsRing ≈ + * - 0# 1# → IsRing ≈ + (flip *) - 0# 1#
+  isRing r = record
+    { +-isAbelianGroup = r-isAbelianGroup
+    ; *-cong = preserves₂ ≈ ≈ ≈ r.*-cong
+    ; *-assoc = associative ≈ * sym r.*-assoc
+    ; *-identity = identity ≈ * r.*-identity
+    ; distrib = distributes ≈ * + r.distrib
+    ; zero = zero ≈ * r.zero
+    }
+    where
+      module r = IsRing r
+      r-isAbelianGroup = r.+-isAbelianGroup
+      open IsAbelianGroup r-isAbelianGroup using (sym)
```

Comment:
Why can we not just use `r.sym`?

Reply:
It seems that we can. This shows my continued failure to understand the intricacies of how each structure/bundle systematically re-exports all its substructures/bundles as 'flat' definitions (always assuming that that is indeed what is going on...)